src/poetry/core/semver/version.py

```diff
@@ -86,12 +86,6 @@ def allows(self, version: Version | None) -> bool:
         # allow weak equality to allow `3.0.0+local.1` for `3.0.0`
         if not _this.is_local() and _other.is_local():
             _other = _other.without_local()
-        elif _this.is_local() and not _other.is_local():
-            _this = _this.without_local()
-
-        # allow weak equality to allow `3.0.0-1` for `3.0.0`
-        if not _this.is_postrelease() and _other.is_postrelease():
```

Comment:
Doesn't the pep only talk about when `_this.is_postrelease()`?

Reply:
I interpret the example

> == 1.1 # Not equal, so 1.1.post1 does not match clause

from pep as follows:

`_this` is 1.1 (no post release)
`_other` is 1.1.post1 (post release)

`_this` should not allow `_other`

Reply:
> For example, given the version 1.1.post1, the following clauses would match or not as shown

Seems to imply `_this` is `1.1.post1` and `_other` is `1.1`.

Reply:
I disagree. In my understanding, the "given version" is the version to be allowed or not (`_other`). The "clause" can be any version constraint (exact version, version range, version union). The clause allows or doesn't allow a given version (clause = `_this`, given version = `_other`).

By the way, considering the third example, it wouldn't make sense to check if `1.1.post1` allows `1.1.*`. Thus, it has to be the other way around, hasn't it?

Reply:
In this particular case it doesn't much matter which reading is right, because the other is also given ten lines later.  ie we have both

> == 1.1        # Not equal, so 1.1.post1 does not match clause

and 

> == 1.1.post1  # Not equal (post-release), so 1.1 does not match clause

Reply:
I stand corrected.